Core/Scenes/TestScene.json

```diff
@@ -0,0 +1,78 @@
+{
+  "assets": [
+    {
+      "folder": "TestAE2/",
+      "file": "ae86.gltf"
+    }
+  ],
+  "entities": [
+    {
+      "name": "Main Camera",
+      "transform": {
+        "localPosition": [ 0, 1, 0 ],
+        "localOrientation": [],
+        "localScale": []
+      },
+      "components": [
+        {
+          "type": "Camera",
+          "params": {
+            "fieldOfView": 90,
+            "farClipPlane": 1000,
+            "nearClipPlane": 0.3
+          }
+        },
+        {
+          "type": "Script",
+          "params": {
+            "scripts": [
+              {
+                "script": "OrbitScript",
+                "params": {
+                  "target": "Test Car",
+                  "m_Radius": 2,
+                  "m_Speed": 1.5,
+                  "currentAngle": 0
+                }
+              }
+            ]
+          }
+        }
+      ]
+    },
+    {
+      "name": "Light",
+      "transform": {
+        "localPosition": [ 0, 5, 0 ],
+        "localOrientation": [ -0.5, 0.5, 0.5, 0 ],
+        "localScale": []
+      },
+      "components": [
+        {
+          "type": "Light",
+          "params": {
+            "color": [ 1.0, 1.0, 1.0, 1.0 ],
+            "intensity": 1,
+            "constant": 0,
+            "linear": 0.0,
+            "quadratic": 1.0,
+            "cutoff": 12.0,
+            "outercutoff": 15.0,
+            "type": "AmbientLight"
+          }
+        }
+      ]
+    },
+    {
+      "name": "Test Car",
+      "transform": {
+        "localPosition": [],
+        "localOrientation": [],
+        "localScale": []
+      },
+      "components": {
+
+      }
+    }
+  ]
+}
```

Comment:
Looks good, nice work, how would parent/ child objects work in this file structure? 

If a scene becomes way more complex how do we keep modifying a scene file simple and easy to do. If we miss parameters will the scene still load?

Does the sceneparser get_to() fail gracefully or does it crash the scene if we don't have one of the params?

Reply:
Also comment 2: 

So if we have a GLTF file for the map, and it has a nested structure for meshes etc, do we have to recreate that scene structure inside a json file and parse them simultaneously? 

Could you break down how, in practice, we might load a scene for the main race track in the main game from load to game ready

Reply:
>how would parent/ child objects work in this file structure?

Parent and child keys have been added. Parent is a single string of the entity that is the parent, created if not found.
Likewise for children as a string array.


>If a scene becomes way more complex how do we keep modifying a scene file simple and easy to do. If we miss parameters will the scene still load?
>
> Does the sceneparser get_to() fail gracefully or does it crash the scene if we don't have one of the params?

Exception handling has been implemented for both .at() and get_to() calls. This will allow some parameters to be optional.

> So if we have a GLTF file for the map, and it has a nested structure for meshes etc, do we have to recreate that scene structure inside a json file and parse them simultaneously?

No. All assets will be loaded first, including the GLTF map file. If any of those entites then have additional components outside of Mesh or Rigidbodies, then they need to be defined within the json.